provider.go

```diff
@@ -1,47 +1,182 @@
-// Package libdnstemplate implements a DNS record management client compatible
-// with the libdns interfaces for <PROVIDER NAME>. TODO: This package is a
-// template only. Customize all godocs for actual implementation.
-package libdnstemplate
+package autodns
 
 import (
 	"context"
 	"fmt"
+	"net/http"
+	"slices"
 
 	"github.com/libdns/libdns"
 )
 
-// TODO: Providers must not require additional provisioning steps by the callers; it
-// should work simply by populating a struct and calling methods on it. If your DNS
-// service requires long-lived state or some extra provisioning step, do it implicitly
-// when methods are called; sync.Once can help with this, and/or you can use a
-// sync.(RW)Mutex in your Provider struct to synchronize implicit provisioning.
+const (
+	autoDNSendpoint string = "https://api.autodns.com/v1"
+	autoDNScontext  string = "4"
+)
 
-// Provider facilitates DNS record manipulation with <TODO: PROVIDER NAME>.
+// Provider facilitates DNS record manipulation with Autodns.
 type Provider struct {
-	// TODO: put config fields here (with snake_case json
-	// struct tags on exported fields), for example:
-	APIToken string `json:"api_token,omitempty"`
+	Username   string `json:"username"`
+	Password   string `json:"password"`
+	Endpoint   string `json:"Endpoint"`
+	Context    string `json:"context"`
+	Primary    string `json:"primary"`
+	HttpClient *http.Client
```

Comment:
```suggestion
	HTTPClient *http.Client `json:"-"`
```

This casing is more conventional (initialisms are treated like this) -- but not required. It should also be excluded from JSON marshalling, I would be surprised if it successfully encoded.